environment.yml

```diff
@@ -6,15 +6,15 @@ channels:
 dependencies:
 - python=3.10
 - pip
-- pydantic>=1.10,<2
+- pydantic>=2.7.3
```

Comment:
Looks mostly fine, though maybe we should pin it to below three?  not sure what their timelines are like but would also not be keen on this randomly breaking.   The difficulty with that arises from the other deps that depend on pydantic, as their versioning isnt as straight forward(check wheel content specifically)